router/keeper/keeper.go

```diff
@@ -33,6 +33,33 @@ type Keeper struct {
 	distrKeeper    types.DistributionKeeper
 }
 
+type PacketMetadata struct {
+	Forward *ForwardMetadata `json:"forward"`
+}
+
+type ForwardMetadata struct {
+	Receiver string          `json:"receiver"`
+	Port     string          `json:"port"`
+	Channel  string          `json:"channel"`
+	Timeout  time.Duration   `json:"timeout"`
+	Retries  *uint8          `json:"retries"`
```

Comment:
Why does the retries field need to be a pointer to an `uint8` vs. an `uint8`?

Reply:
The chain can set the default retries (could be non-zero), and the packet can override it. This makes sure that if `retries` is not present in the json, it will be nil (and should use the chain default) instead of 0.